pkg/validation/internal/object.go

```diff
@@ -0,0 +1,199 @@
+package internal
+
+import (
+	"encoding/json"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var ObjectValidator interfaces.Validator = interfaces.ValidatorFunc(validateObjects)
+
+const (
+	PodDisruptionBudgetKind     = "PodDisruptionBudget"
+	PriorityClassKind           = "PriorityClass"
+	RoleKind                    = "Role"
+	ClusterRoleKind             = "ClusterRole"
+	PodDisruptionBudgetAPIGroup = "policy"
+	SCCAPIGroup                 = "security.openshift.io"
+)
+
+var defaultSCCs = map[string]struct{}{
+	"privileged":       {},
+	"restricted":       {},
+	"anyuid":           {},
+	"hostaccess":       {},
+	"hostmount-anyuid": {},
+	"hostnetwork":      {},
+	"node-exporter":    {},
+	"nonroot":          {},
+}
+
+func validateObjects(objs ...interface{}) (results []errors.ManifestResult) {
+	for _, obj := range objs {
+		switch u := obj.(type) {
+		case *unstructured.Unstructured:
```

Comment:
This is more a general comment about how validators are set up than this particular function, but the validator caller has to know what type to pass to this internal function, which isn't great. We may want to rethink the signature of the validator interface for this reason.

Reply:
I agree. Generally all `kubectl-able` objects fulfill the `Unstructured` interface, so I think it's ok for now, but as you said it isn't flexible in what it accepts to validate.

Reply:
`unstructured.Unstructured` isn't an interface, it's a type. Do you mean `runtime.Object`?

Reply:
Yep, that's what I meant

I propose we keep this as-is for now and revisit the signature of the validators as part of a separate PR